specification/azure-kusto/resource-manager/Microsoft.Kusto/stable/2019-11-09/kusto.json

```diff
@@ -0,0 +1,3986 @@
+            "schema": {
+              "$ref": "#/definitions/CloudError"
+            }
+          }
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Kusto/clusters/{clusterName}/principalAssignments/{principalAssignmentName}": {
+      "get": {
+        "tags": [
+          "ClusterPrincipalAssignments"
+        ],
+        "operationId": "ClusterPrincipalAssignments_Get",
+        "x-ms-examples": {
+          "KustoClusterPrincipalAssignmentsGet": {
+            "$ref": "./examples/KustoClusterPrincipalAssignmentsGet.json"
+          }
+        },
+        "description": "Gets a Kusto cluster principalAssignment.",
+        "parameters": [
+          {
+            "$ref": "#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "#/parameters/ResourceGroupParameter"
+          },
+          {
+            "$ref": "#/parameters/ClusterNameParameter"
+          },
+          {
+            "$ref": "#/parameters/PrincipalAssignmentNameParameter"
+          },
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "The Kusto cluster principal assignment object.",
+            "schema": {
+              "$ref": "#/definitions/ClusterPrincipalAssignment"
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "#/definitions/CloudError"
+            }
+          }
+        }
+      },
+      "put": {
+        "tags": [
+          "ClusterPrincipalAssignments"
+        ],
+        "operationId": "ClusterPrincipalAssignments_CreateOrUpdate",
+        "x-ms-examples": {
+          "KustoClusterPrincipalAssignmentsCreateOrUpdate": {
+            "$ref": "./examples/KustoClusterPrincipalAssignmentsCreateOrUpdate.json"
+          }
+        },
+        "x-ms-long-running-operation": true,
+        "description": "Create a Kusto cluster principalAssignment.",
+        "parameters": [
+          {
+            "$ref": "#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "#/parameters/ResourceGroupParameter"
+          },
+          {
+            "$ref": "#/parameters/ClusterNameParameter"
+          },
+          {
+            "$ref": "#/parameters/PrincipalAssignmentNameParameter"
+          },
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          },
+          {
+            "name": "parameters",
+            "in": "body",
+            "required": true,
+            "schema": {
+              "$ref": "#/definitions/ClusterPrincipalAssignment"
+            },
+            "description": "The Kusto cluster principalAssignment's parameters supplied for the operation."
+          }
+        ],
+        "responses": {
+          "200": {
```

Comment:
I see you've declared `x-ms-long-running-operation` - can this API return 202 then, or will it always complete synchronously?

Reply:
This is an asynchronous API, but put never returns 202. Please check "Clusters_CreateOrUpdate"